### PR TITLE
PufferDepositor and access

### PR DESCRIPTION
**Pull request by bxmmm1**
- Remove unused code from the PufferDepositor
- Create a helper script for access management

### CODE REVIEW

File: script/GenerateAccessManagerCallData.sol [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+contract GenerateAccessManagerCallData is Script {
+    function run(address pufferVaultProxy) public pure returns (bytes memory) {
+        bytes[] memory calldatas = new bytes[](2);
+
+        // Public selectors
+        bytes4[] memory publicSelectors = new bytes4[](6);
+        publicSelectors[0] = PufferVaultMainnet.withdraw.selector;
+        publicSelectors[1] = PufferVaultMainnet.redeem.selector;
+        publicSelectors[2] = PufferVaultMainnet.redeem.selector;
```
Comment (CheyenneAtapour) [inline script/GenerateAccessManagerCallData.sol:28]: Why 2 selectors for redeem?

File: src/PufferVaultMainnet.sol [hunk 2]
```diff
@@ -189,6 +228,8 @@ contract PufferVaultMainnet is PufferVault {
 
     /**
      * @notice Allows the `msg.sender` to burn his shares
+     * @dev Restricted to PufferProtocol smart contract
+     * We use it to burn the bond if the node operator gets slashed
      * @param shares The amount of shares to burn
      */
     function burn(uint256 shares) public restricted {
```
Comment (CheyenneAtapour) [inline src/PufferVaultMainnet.sol:235]: Nbd, but we mentioned we could allow anyone to call this to basically make a donation. The alternative they'd have is to send to something like address(0), but doesn't really burn the token / change the exchange ratio. IIRC we discussed making this pausable

**Review by CheyenneAtapour — CHANGES_REQUESTED** [review]
duplicate selector and public burn (see comments)

File: src/PufferVaultMainnet.sol (hunk 2 above)
Comment (bxmmm1) [reply src/PufferVaultMainnet.sol:235]: Less things to think about. If somebody wants to donate, they can do so by depositing ETH/stETH/WETH

**Pull request merged**